app/controllers/api/v4/card_grants_controller.rb

```diff
@@ -16,7 +16,19 @@ def index
       def create
         @event = Event.find_by_public_id(params[:event_id]) || Event.friendly.find(params[:event_id])
 
-        @card_grant = @event.card_grants.build(params.permit(:amount_cents, :email, :merchant_lock, :category_lock, :keyword_lock, :purpose, :one_time_use, :pre_authorization_required, :instructions).merge(sent_by: current_user))
+        @current_user = current_user
+        if current_user.admin? && params[:sent_by_email].present?
+          found_user = User.find_by(email: params[:sent_by_email].strip.downcase)
+
+          if found_user.nil?
+            skip_authorization
+            return render json: { error: "invalid_user", messages: "User with email '#{params[:sent_by_email]}' not found" }, status: :bad_request
+          end
+
+          @current_user = found_user
+        end
+
+        @card_grant = @event.card_grants.build(params.permit(:amount_cents, :email, :merchant_lock, :category_lock, :keyword_lock, :purpose, :one_time_use, :pre_authorization_required, :instructions).merge(sent_by: @current_user))
```

Comment:
I think it's a bit dangerous to muck with the `@current_user` variable. I understand we want to say that card grant was issued by `found_user`, however, changing `@current_user` would imply that the current user of this entire request is `found_user`, which isn't entirely true.

The reason is because the `current_user` method uses `@current_user` under the hood to store who the current user is.

```suggestion
        sent_by = current_user

        if current_user.admin? && params[:sent_by_email].present?
          found_user = User.find_by(email: params[:sent_by_email].strip.downcase)

          if found_user.nil?
            skip_authorization
            return render json: { error: "invalid_user", messages: "User with email '#{params[:sent_by_email]}' not found" }, status: :bad_request
          end

          sent_by = found_user
        end

        @card_grant = @event.card_grants.build(params.permit(:amount_cents, :email, :merchant_lock, :category_lock, :keyword_lock, :purpose, :one_time_use, :pre_authorization_required, :instructions).merge(sent_by:))
```